packages/cashier/src/Stores/general-store.js

```diff
@@ -217,11 +222,13 @@ export default class GeneralStore extends BaseStore {
 
     // Initialise P2P attributes on app load without mounting the entire cashier
     @action.bound
-    init() {
+    async init() {
         if (this.root_store.modules.cashier) {
-            const { client, modules } = this.root_store;
+            const {
+                client: { is_logged_in, switched },
+                modules,
+            } = this.root_store;
             const { account_prompt_dialog, withdraw } = modules.cashier;
-            const { currency, is_logged_in, switched } = client;
 
             when(
```

Comment:
@vinu-deriv I'm thinking of omitting this `when` as well.
`when(
                () => is_logged_in,
                async () => {
                    await this.getAdvertizerError();
                    this.checkP2pStatus();
                    await withdraw.check10kLimit();
                }
            );
`
and move this `await this.getAdvertizerError();` to after `await this.WS.wait('get_settings');`
What do you think?